File: SelectionSort.cpp
#include<iostream>
using namespace std;

class SelectionSort {

	public:
	
	int sort(int arr[], int i){
		
		for(int left = i; left >= 0; left--){
			if(arr[i] < arr[left]){		
				i = left;
			}
		}	
		return i;
	}
		
	void swap(int *xp, int *yp){
	    int temp = *xp;
	    *xp = *yp;
	    *yp = temp;
	}
	
	void select(int arr[], int n) {
	   int i, j;
	   for (i = n-1; i >=0; i--)     {
	        int j = sort(arr, i);     
	        swap(&arr[i], &arr[j]);
	   }
	      
	   for (i = 0; i < n; i++)     {
	    	cout << arr[i] << "\t";
	   }
	}
};


int main(){
	
	int array[6] = {22, 4, 1, 55, 10, 98};
	
	SelectionSort obj;
	obj.select(array, 6);
	
	return 0;
}

File: Sorting/SelectionSort.java
import java.util.Scanner;
public class SelectionSort {
	
	public void sort(int arr[]){
		for(int i=0; i<arr.length; i++){
			int j = i;
			
			for(int k=i+1; k<arr.length; k++){
				if(arr[j] > arr[k]){
					j = k;
				}
			}
			
			int temp = arr[i];
			arr[i] = arr[j];
			arr[j] = temp;
			
			printArray(arr);
		}
	}
	
	public void printArray(int arr[]){
		for(int i=0; i<arr.length; i++){
			System.out.print(arr[i]+" ");
		}
		System.out.println();
	}
	
	public static void main(String...strings){
		
		SelectionSort obj = new SelectionSort();
		Scanner in = new Scanner(System.in);
		
		int size = in.nextInt();
		int[] arr = new int[size];
		
		for(int i=0; i<size; i++){
			arr[i] = in.nextInt();
		}
		obj.sort(arr);
		in.close();
	}
}
